app/models/validations/sales/household_validations.rb

```diff
@@ -20,6 +20,27 @@ def validate_previous_postcode(record)
     end
   end
 
+  def validate_buyers_living_in_property(record)
+    return unless record.form.start_date.year >= 2023
+    return unless record.buylivein
+
+    if record.buyers_will_live_in?
+      if record.joint_purchase? && record.buyer_one_will_not_live_in_property? && record.buyer_two_will_not_live_in_property?
+        record.errors.add :buylivein, I18n.t("validations.household.buyers_will_live_in_property.buyers_live_but_no_buyers_live.joint.buyers")
+        record.errors.add :buy1livein, I18n.t("validations.household.buyers_will_live_in_property.buyers_live_but_no_buyers_live.joint.each_buyer")
+        record.errors.add :buy2livein, I18n.t("validations.household.buyers_will_live_in_property.buyers_live_but_no_buyers_live.joint.each_buyer")
+      elsif record.not_joint_purchase? && record.buyer_one_will_not_live_in_property?
+        record.errors.add :buylivein, I18n.t("validations.household.buyers_will_live_in_property.buyers_live_but_no_buyers_live.not_joint.buyers")
+        record.errors.add :jointpur, I18n.t("validations.household.buyers_will_live_in_property.buyers_live_but_no_buyers_live.not_joint.jointpur")
+        record.errors.add :buy1livein, I18n.t("validations.household.buyers_will_live_in_property.buyers_live_but_no_buyers_live.not_joint.each_buyer")
+      end
+    elsif record.buyer_one_will_live_in_property? || record.buyer_two_will_live_in_property?
```

Comment:
I had 37-41 as an else with a separate if inside it which I think read a little more clearly as the branch for when buyers_live_in? is false, but the linter made me collapse the nested if